### PR TITLE
Managed instance updates plus examples

### PR DESCRIPTION
**Pull request by johnpaulkee**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json [hunk 1]
```diff
@@ -361,7 +430,7 @@
           ]
         },
         "dnsZone": {
-          "description": "The Dns Zone that the managed instance is in.",
+          "description": "The Dns Zone taht the managed instance is in.",
```
Comment (nschonni) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]:
```suggestion
          "description": "The Dns Zone that the managed instance is in.",
```

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json [hunk 2]
```diff
@@ -376,22 +445,36 @@
...
-          "description": "Id of the timezone. Allowed values are timezones supported by Windows.\r\nWindows keeps details on supported timezones, including the id, in registry under\r\nKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Time Zones.\r\nYou can get those registry values via SQL Server by querying SELECT name AS timezone_id FROM sys.time_zone_info.\r\nList of Ids can also be obtained by executing [System.TimeZoneInfo]::GetSystemTimeZones() in PowerShell.\r\nAn example of valid timezone id is \"Pacific Standard Time\" or \"W. Europe Standard Time\".",
+          "description": "Id of the timezone. Allowed values are timezones supported by Windows.\r\nWinodws keeps details on supported timezones, including the id, in registry under\r\nKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Time Zones.\r\nYou can get those registry values via SQL Server by querying SELECT name AS timezone_id FROM sys.time_zone_info.\r\nList of Ids can also be obtained by executing [System.TimeZoneInfo]::GetSystemTimeZones() in PowerShell.\r\nAn example of valid timezone id is \"Pacific Standard Time\" or \"W. Europe Standard Time\".",
```
Comment (nschonni) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]:
```suggestion
          "description": "Id of the timezone. Allowed values are timezones supported by Windows.\r\nWindows keeps details on supported timezones, including the id, in registry under\r\nKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Time Zones.\r\nYou can get those registry values via SQL Server by querying SELECT name AS timezone_id FROM sys.time_zone_info.\r\nList of Ids can also be obtained by executing [System.TimeZoneInfo]::GetSystemTimeZones() in PowerShell.\r\nAn example of valid timezone id is \"Pacific Standard Time\" or \"W. Europe Standard Time\".",
```

**Review by nschonni — COMMENTED** [review]
Not sure if this is generated, but it's reintroducing fixed typos

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5899 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5899/commits/bc8da6dd3e0856e9b8e8af11c6298afc663ffd6f

**Review by jaredmoo — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json (hunk 1 above)
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: > Not sure if this is generated, but it's reintroducing fixed typos

Yup these were generated, I'll make some fixes in our examples internally and update.
Thanks @nschonni for spotting these.

**Review by petrajkogit — APPROVED** [review]

**Comment by petrajkogit** [comment]
Thanks for this PR!

**Comment by johnpaulkee** [comment]
> Thanks for this PR!

Np :) Thanks for reviewing @jaredmoo and @petrajkogit !

**Comment by johnpaulkee** [comment]
@nschonni , not sure what's going on with the build automation for azure-sdk-for-net.
is this a transient build err?
```
19:34:24          StartInfo: '/c autorest.cmd https://raw.githubusercontent.com/johnpaulkee/azure-rest-api-specs/f75801138eba6842afbd124547f47aa66e4383a1/specification/sql/resource-manager/readme.md --csharp --version=2.0.4283 --reflect-api-versions --csharp.output-folder=C:\jenkins\workspace\DotNetSwaggerToSdkCI\tmp2C5E\src\SDKs\SqlManagement\Management.Sql\Generated'
19:44:24      1>c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error : InvalidOperationException: Process must exit before requested information can be determined.
19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error :    at System.Diagnostics.Process.EnsureState(State state)
19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error :    at System.Diagnostics.Process.get_ExitCode()
19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error :    at MS.Az.SwaggerToSdk.Common.ExecProcess.ShellExec.ExecuteCommand(String args, String workingDir) in F:\tf\AzNetSdkTools\src\SwaggerToSdk\Common\SToS.Common\ExecProcess\ShellExec.cs:line 147
19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error : 
19:44:25          StartInfo: '/target:library /out:C:\Users\FXSign\AppData\Local\Temp\FSUtil\tmpC7D3\SqlManagement-SdkInfo.dll C:\jenkins\workspace\DotNetSwaggerToSdkCI\tmp2C5E\src\SDKs\SqlManagement\Management.Sql\Generated\SdkInfo_SqlManagementClient.cs'
19:44:26          Microsoft (R) Visual C# Compiler version 2.10.0.0 (b9fb1610)
19:44:26          Copyright (C) Microsoft Corporation. All rights reserved.
```

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json [hunk 3]
```diff
@@ -308,6 +354,21 @@
       "description": "The properties of a managed instance.",
       "type": "object",
       "properties": {
+        "managedInstanceCreateMode": {
+          "description": "Specifies the mode of database creation.\r\n\r\nDefault: Regular instance creation.\r\n\r\nRestore: Creates an instance by restoring a set of backups to specific point in time. RestorePointInTime and SourceManagedInstanceId must be specified.",
```
Comment (kpajdzik) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: I think `\n` would be enough but that's alright.

**Review by kpajdzik — APPROVED** [review]

**Comment by kpajdzik** [comment]
> @nschonni , not sure what's going on with the build automation for azure-sdk-for-net.
> is this a transient build err?
> 
> ```
> 19:34:24          StartInfo: '/c autorest.cmd https://raw.githubusercontent.com/johnpaulkee/azure-rest-api-specs/f75801138eba6842afbd124547f47aa66e4383a1/specification/sql/resource-manager/readme.md --csharp --version=2.0.4283 --reflect-api-versions --csharp.output-folder=C:\jenkins\workspace\DotNetSwaggerToSdkCI\tmp2C5E\src\SDKs\SqlManagement\Management.Sql\Generated'
> 19:44:24      1>c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error : InvalidOperationException: Process must exit before requested information can be determined.
> 19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error :    at System.Diagnostics.Process.EnsureState(State state)
> 19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error :    at System.Diagnostics.Process.get_ExitCode()
> 19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error :    at MS.Az.SwaggerToSdk.Common.ExecProcess.ShellExec.ExecuteCommand(String args, String workingDir) in F:\tf\AzNetSdkTools\src\SwaggerToSdk\Common\SToS.Common\ExecProcess\ShellExec.cs:line 147
> 19:44:24 c:\jenkins\workspace\DotNetSwaggerToSdkCI\scripts\stostools\SToS.proj(17,9): error : 
> 19:44:25          StartInfo: '/target:library /out:C:\Users\FXSign\AppData\Local\Temp\FSUtil\tmpC7D3\SqlManagement-SdkInfo.dll C:\jenkins\workspace\DotNetSwaggerToSdkCI\tmp2C5E\src\SDKs\SqlManagement\Management.Sql\Generated\SdkInfo_SqlManagementClient.cs'
> 19:44:26          Microsoft (R) Visual C# Compiler version 2.10.0.0 (b9fb1610)
> 19:44:26          Copyright (C) Microsoft Corporation. All rights reserved.
> ```

This failure looks transient. I'll ping .NET generator owners.

**Comment by kpajdzik** [comment]
@shahabhijeet, Could you take a look at .NET failures?

**Comment by johnpaulkee** [comment]
Let me know if the azure-sdk-for-net issue is a problem to fix before merge. We're ready on our end if it isn't a problem.
Also, I just retried build and we're getting the same err.

**Comment by dsgouda** [comment]
@johnpaulkee @kpajdzik Please disregard the .NET failure here, we have some OOM exceptions when compiling SQL SDK given its size. Please feel free to approve and merge

**Pull request merged**